src/app/SharpRaven/Data/JsonPacket.cs

```diff
@@ -123,7 +123,16 @@ private static object Merge(SentryEvent @event)
                 }
             }
             else
-                result = JObject.FromObject(extra);
+            {
+                try
+                {
+                    result = JObject.FromObject(extra);
+                }
+                catch (ArgumentException e)
+                {
+                    result = JObject.Parse(string.Format(@"{{""{0}"":""{1}""}}", extra.GetType(), extra));
```

Comment:
It seems the solution is either this or `dynamic` object. I personally do not favor any usage of `dynamic`, though.